R/mzQCjson.R

```diff
@@ -0,0 +1,255 @@
+  }                                                                                                                               }
+  
+  ##delete the row where the value of data is NA 
+  data <- data[! is.na(data$value), ]
+  if(runQuality == TRUE) data[length(data$quality_type) + 1, ] <- c(NA, NA, NA, NA, NA, "runQuality", "Raw file: Ecoli")
+  if(setQuality == TRUE) data[length(data$quality_type) + 1, ] <- c(NA, NA, NA, NA, NA, "setQuality", "Raw file: Ecoli")
+  
+  return(data)
+}
+
+#'
+#' Extract the data of each row in data and store it in a list
+#' 
+#' 
+#' 
+#' @param data A list 
+#' @return a list
+#' 
+#' @export
+#' 
+#' 
+#' 
+transform_data <-function(data){
+  trans <- list()
+  for (x in (1:length(data$cvRef))) {
+    trans[x] = list(data[x, ])
+  }
+  return(trans)
+}
+  
+#'
+#' Get the Metadata composed of inputFiles and analysisSoftware
+#' 
+#' 
+#' 
+#' @param input Raw file of metric data
+#' @param data_input File based on which QC metrics were generated
+#'                   use analysisSofteware to connect data_input and data_analysis
+#' @param data_analysis Software tools used to generate the QC metrics
+#' @return a list
+#' 
+#' @export
+#' 
+#' 
+#' 
+getmetaData <- function(input, data_input, data_analysis){
+  inputFile <- list()
+  analysis <- list()
+  input <- unique(input)
+  for (x in (1:length(input))) {
+    locate <- which(input[x] == data_input$name)
+    inputFile[x] <- list(data_input[locate, ][1:3])
+    analysis[x] <- list(data_analysis[which(data_input[locate,]$analysisSoftware == data_analysis$name), ])
+  }
+  analysis <- unique(analysis)
+  
+  metaData = list(inputFiles = list(inputFile = inputFile), analysisSoftware = analysis)
+  return(metaData)
+}
+
+#'
+#' Get the Qualities(either runQualities or setQualities) composed of metadata and qualityMetrics
+#' 
+#' 
+#' 
+#' @param Qualities the filtered data frame of metric
+#' @param data_input File based on which QC metrics were generated
+#'                   use analysisSofteware to connect data_input and data_analysis
+#' @param data_analysis Software tools used to generate the QC metrics
+#' @param set judge whether the quality is setQuality
+#' @return a list
+#' 
+#' @export
+#' 
+#' 
+#' 
+getQualities <- function(Qualities, data_input, data_analysis, set){
+  if(length(Qualities[ , 1]) != 0 ){
+    if(set == TRUE){
+      input <- as.list(Qualities$input_file)
+      input_file <- unlist(lapply(input, function(x){
+        if (grepl(",", x)){
+          input[which(x == as.list(Qualities$input_file))] <- NULL
+          input <- unlist(c(input, unlist(strsplit(x , split = ","))))
+        }
+      }))
+    }else{
+      input_file <- as.list(Qualities$input_file)
+    }
+    metaData <- getmetaData(input = input_file, data_input, data_analysis)
+    Qualities <- Qualities[! is.na(Qualities$value), ]
+    Qualities <- list(metaData = metaData, qualityMetrics = list(qualityMetric = transform_data(data = Qualities[1:5])))
```

Comment:
why 1:5?

Reply:
Because metric_info has seven columns, "cvRef", "accession","name","unit","value","quality_type","input_file", but in cvParameter I only need the information for the first five columns

Reply:
ah ok. 
Just name the columns which you need using strings.
That is much more intuitive.